Backend/app/loan_management/manage_loans_service.py

```diff
@@ -250,24 +250,75 @@ def _check_reserved_book(
             )
         return None
 
-    def consult_limit_by_user_email(self, email: str) -> bool:
+    def cant_reserved_loans(self, email: str) -> int:
         cant_reserved_loans = self.query_database(
             """SELECT COUNT(*)
                 FROM loan
                 WHERE userEmail = ?
                 AND loanStatus = ? AND returnDate IS NULL""",
             (email, "reserved"),
         )
+        return cant_reserved_loans[0]
+
+    def cant_loaned_books(self, email: str) -> int:
         cant_loaned_books = self.query_database(
             """SELECT COUNT(*)
                 FROM loan
                 WHERE userEmail = ?
                 AND loanStatus = ? AND returnDate IS NULL""",
             (email, "loaned"),
         )
+        return cant_loaned_books[0]
+
+    def cant_points_user(self, email: str) -> int:
+        points_user = self.query_database(
+            """SELECT points FROM users WHERE email= ?""", (email,)
+        )
+        return points_user[0]
+
+    def limit_amount_of_loans(
+        self,
+        email: str,
+        points_user: int,
+        cant_reserved_loans: int,
+        cant_loaned_books: int,
+    ):
 
-        # TODO: fix
-        if cant_reserved_loans[0] >= 3 or cant_loaned_books[0] >= 3:
+        is_limit_default: bool = cant_reserved_loans == 3 or cant_loaned_books == 3
+
+        if points_user < 100 and is_limit_default:
+            self.execute_in_database(
+                """UPDATE users SET loanLimit = ? WHERE email = ?""",
+                (3, email),
+            )
+        if points_user >= 100:
+            self.execute_in_database(
+                """UPDATE users SET loanLimit = ? WHERE email = ?""",
+                (5, email),
+            )
+
+    def consult_limit_by_user_email(self, email: str) -> bool:
+
+        points_user = self.cant_points_user(email)
+        cant_reserved_loans = self.cant_reserved_loans(email)
+        cant_loaned_books = self.cant_loaned_books(email)
+
+        self.limit_amount_of_loans(
+            email, points_user, cant_reserved_loans, cant_loaned_books
+        )
+
+        limit_user = self.query_database(
+            """SELECT loanLimit FROM users WHERE email= ?""", (email,)
+        )
+        loan_limit = limit_user[0]
+
+        is_limit_reached = (
+            cant_reserved_loans >= loan_limit or cant_loaned_books >= loan_limit
+        )
+
+        if loan_limit == 3 and is_limit_reached and points_user < 100:
```

Comment:
All of these comparisons are unnecessary. There's no reason to check user points at this point. You can simply:
```py
return cant_loaned_or_reserved <= loan limit
```